actionview/lib/action_view/template/handlers/erb.rb

```diff
@@ -1,68 +1,65 @@
-require "erubis"
+require "erubi"
 
 module ActionView
   class Template
     module Handlers
-      class Erubis < ::Erubis::Eruby
-        def add_preamble(src)
+      class Erubi < ::Erubi::Engine
+        def initialize(input, properties = {})
           @newline_pending = 0
-          src << "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+
+          properties = Hash[properties]
+          properties[:preamble] = "@output_buffer = output_buffer || ActionView::OutputBuffer.new;"
+          properties[:postamble] = "@output_buffer.to_s"
+          properties[:bufvar] = "@output_buffer"
+          properties[:escapefunc] = ""
+          super
         end
 
-        def add_text(src, text)
+        def evaluate(_context)
```

Comment:
Shadowing generally shows up in method names rather than local names, because the latter are only relevant when eval/binding gets involved. For methods, yes, we do use an underscore. I don't recall whether we have other places that care about locals.